client-network/src/network_ops/default_network_ops_client.rs

```diff
@@ -166,6 +166,15 @@ where
             .iter()
             .map(|(input, _)| input.clone())
             .collect::<Vec<_>>();
+
+        let staked_state = self.get_staked_state(&to_address)?;
```

Comment:
I think `get_staked_state` return error if address not exists on chain, but should be ok to be the to_address of deposit tx.

Reply:
For withdraw, it's ok.
For deposit, it could be the case that this staked state wasn't created in genesis and this is the first transaction

Reply:
When deposit, if the staking address belongs to the wallet and `staked_state` does not exist, we let it go so that the user can be a validator. In other cases, if the staking address not exists, we can prevent a user's loss.

Reply:
> When deposit, if the staking address belongs to the wallet and `staked_state` does not exist, we let it go so that the user can be a validator. In other cases, if the staking address not exists, we can prevent a user's loss.

I think this function is about deposit tx, so it's problematic to return error on non-exist staking address.

Reply:
>  In other cases, if the staking address not exists, we can prevent a user's loss.

it may be the case that it's really what you want to do -- e.g. @lezzokafka frequently deposits to addresses that are not managed by his wallet to distribute testnet tokens .

so rather than preventing it, it may just have an extra confirmation step